### PR TITLE
[FSDPv2] Add a user guide

### PR DESCRIPTION
**Pull request by alanwaketan**
Summary:
This diff adds a user guide for FSDPv2.

### CODE REVIEW

**Comment by JackCaoG** [comment]
Can you also update https://github.com/pytorch/xla/blob/master/docs/source/index.rst so it will show up in https://pytorch.org/xla/release/2.1/index.html?

File: docs/fsdpv2.md [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+mesh = Mesh(device_ids, mesh_shape, ('fsdp', 'model'))
+
+# As normal FSDP, but an extra mesh is needed.
+model = FSDPv2(my_module, mesh)
+optim = torch.optim.Adam(model.parameters(), lr=0.0001)
+output = model(x, y)
+loss = output.sum()
+loss.backward()
+optim.step()
+```
```
Comment (JackCaoG) [inline docs/fsdpv2.md:28]: should we add a note that don't use `xmp.spawn` for this somewhere in this doc?

Comment (alanwaketan) [reply docs/fsdpv2.md:28]: Maybe I just link to the spmd doc? And ask the users to go over it first?

Comment (JackCaoG) [reply docs/fsdpv2.md:28]: depending on your target user. If you are trying to convert those who uses FSDP today to use FSDPv2 I think you want to be more explict about how to migrate their existing code to FSDP v2

File: docs/fsdpv2.md [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+num_devices = xr.global_runtime_device_count()
+mesh_shape = (num_devices, 1)
+device_ids = np.array(range(num_devices))
+# To be noted, the mesh must have an axis named 'fsdp', which the weights and activations will be sharded on.
+mesh = Mesh(device_ids, mesh_shape, ('fsdp', 'model'))
+
+# As normal FSDP, but an extra mesh is needed.
+model = FSDPv2(my_module, mesh)
+optim = torch.optim.Adam(model.parameters(), lr=0.0001)
+output = model(x, y)
```
Comment (jonb377) [inline docs/fsdpv2.md:24]: We'll need to add input sharding here - do you want to add an example with MpDeviceLoader?

File: docs/fsdpv2.md (hunk 1 above)
Comment (alanwaketan) [reply docs/fsdpv2.md:28]: Let's target people that is willing to try SPMD first. Once it's mature, we can then target people that is currently using FSDP.

File: docs/fsdpv2.md (hunk 2 above)
Comment (alanwaketan) [reply docs/fsdpv2.md:24]: Let's manually shard the x here. Good catch!

File: docs/fsdpv2.md (hunk 1 above)
Comment (JackCaoG) [reply docs/fsdpv2.md:33]: sure, it is still good to mention something along the line of "This is using SPMD so you don't need to spawn multiple process per host, for more detail please check this doc"

Comment (alanwaketan) [reply docs/fsdpv2.md:33]: Added one line on top:
Please review the [SPMD user guide](./spmd.md) before proceeding.

**Review by JackCaoG — APPROVED** [review]

**Comment by alanwaketan** [comment]
Thanks, @JackCaoG!
